server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -236,13 +297,21 @@ static Map<String, Object> convertToLoggedFields(List<HealthIndicatorResult> ind
         @Override
         public void onResponse(List<HealthIndicatorResult> healthIndicatorResults) {
             try {
-                Map<String, Object> resultsMap = convertToLoggedFields(healthIndicatorResults);
+                if (outputMode != OutputMode.METRICS) {
```

Comment:
Rather than have this negative check here and one below for `LOGS`, I think it'd be better to push some of that logic into the enum itself. I envision something like:

```java
if (outputMode.logsEnabled()) {
    ... output to logs ...
}
if (outputMode.metricsEnabled()) {
    ... output to metrics ...
}
```

And then the enum implements those methods, which can return true if the enum is set to "all" (aka "both" as this PR is currently). Otherwise we end up making this logic fragile, because if we were to add another output to the enum (`OutputMode.SLACK` for instance), then it would have different behavior and a bug would be introduced.

Reply:
I ended up doing this as a method on `HealthPeriodicLogger` itself since we changed the mode to a list. Let me know if you see a cleaner way though.